packages/in_app_purchase/in_app_purchase_android/android/src/main/java/io/flutter/plugins/inapppurchase/MethodCallHandlerImpl.java

```diff
@@ -144,155 +97,107 @@ void setActivity(@Nullable Activity activity) {
-      case MethodNames.START_CONNECTION:
-        final int handle = (int) call.argument(MethodArgs.HANDLE);
-        int billingChoiceMode = BillingChoiceMode.PLAY_BILLING_ONLY;
-        if (call.hasArgument(MethodArgs.BILLING_CHOICE_MODE)) {
-          billingChoiceMode = call.argument(MethodArgs.BILLING_CHOICE_MODE);
-        }
-        startConnection(handle, result, billingChoiceMode);
-        break;
-      case MethodNames.END_CONNECTION:
-        endConnection(result);
-        break;
-      case MethodNames.QUERY_PRODUCT_DETAILS:
-        List<Product> productList = toProductList(call.argument("productList"));
-        queryProductDetailsAsync(productList, result);
-        break;
-      case MethodNames.LAUNCH_BILLING_FLOW:
-        launchBillingFlow(
-            (String) call.argument("product"),
-            (String) call.argument("offerToken"),
-            (String) call.argument("accountId"),
-            (String) call.argument("obfuscatedProfileId"),
-            (String) call.argument("oldProduct"),
-            (String) call.argument("purchaseToken"),
-            call.hasArgument("prorationMode")
-                ? (int) call.argument("prorationMode")
-                : PRORATION_MODE_UNKNOWN_SUBSCRIPTION_UPGRADE_DOWNGRADE_POLICY,
-            result);
-        break;
-      case MethodNames.QUERY_PURCHASES_ASYNC:
-        queryPurchasesAsync((String) call.argument("productType"), result);
-        break;
-      case MethodNames.QUERY_PURCHASE_HISTORY_ASYNC:
-        queryPurchaseHistoryAsync((String) call.argument("productType"), result);
-        break;
-      case MethodNames.CONSUME_PURCHASE_ASYNC:
-        consumeAsync((String) call.argument("purchaseToken"), result);
-        break;
-      case MethodNames.ACKNOWLEDGE_PURCHASE:
-        acknowledgePurchase((String) call.argument("purchaseToken"), result);
-        break;
-      case MethodNames.IS_FEATURE_SUPPORTED:
-        isFeatureSupported((String) call.argument("feature"), result);
-        break;
-      case MethodNames.GET_CONNECTION_STATE:
-        getConnectionState(result);
-        break;
-      case MethodNames.GET_BILLING_CONFIG:
-        getBillingConfig(result);
-        break;
-      case MethodNames.IS_ALTERNATIVE_BILLING_ONLY_AVAILABLE:
-        isAlternativeBillingOnlyAvailable(result);
-        break;
-      case MethodNames.CREATE_ALTERNATIVE_BILLING_ONLY_REPORTING_DETAILS:
-        createAlternativeBillingOnlyReportingDetails(result);
-        break;
-      case MethodNames.SHOW_ALTERNATIVE_BILLING_ONLY_INFORMATION_DIALOG:
-        showAlternativeBillingOnlyInformationDialog(result);
-        break;
-      default:
-        result.notImplemented();
-    }
-  }
-
-  private void showAlternativeBillingOnlyInformationDialog(final MethodChannel.Result result) {
-    if (billingClientError(result)) {
+  public void showAlternativeBillingOnlyInformationDialog(
+      @NonNull Result<PlatformBillingResult> result) {
+    if (billingClient == null) {
```

Comment:
I think this is a behavior change for "endConnection" It used to be safe to call multiple times aka on a null billing client and with this check is is no longer ok.

Reply:
nvm there is no longer a shared onMethodCall.